DREAM/Optimization/tsgOptimizationUtils.hpp

```diff
@@ -157,6 +157,39 @@ inline double compute_stationarity_residual(const std::vector<double> &x, const
     return std::sqrt(residual);
 }
 
+/*! \ingroup OptimizationUtil
+ * Nesterov's "worst function in the world" for convex optimization. This function is difficult for nearly all first-order
+ * iterative optimization methods starting from x0=0. This generator writes the function and gradient to \b func and \b grad, and
+ * the minimum is written to \b minimum. It is expected that 1 <= k <= (number of dimensions - 1) / 2 and the initial point
+ * of an iterative algorithm is the zero vector.
+ */
+inline void makeNesterovTestFunction(const double L, const int k, ObjectiveFunctionSingle &func, GradientFunctionSingle &grad,
```

Comment:
Are you using this as testing or is it part of the library.

If you want to test with this, make it a function somewhere in testing not in the library.